controllers/configmap_controller.go

```diff
@@ -18,45 +18,243 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/go-logr/logr"
-	"k8s.io/apimachinery/pkg/runtime"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	core "k8s.io/api/core/v1"
+	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/cluster"
+	"github.com/openshift/windows-machine-config-operator/pkg/metrics"
+	"github.com/openshift/windows-machine-config-operator/pkg/nodeconfig"
+	"github.com/openshift/windows-machine-config-operator/pkg/secrets"
+	"github.com/openshift/windows-machine-config-operator/pkg/signer"
+)
+
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=configmaps/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups="",resources=configmaps/finalizers,verbs=update
+
+const (
+	// BYOHAnnotation is an an anotation that should be applied to all Windows nodes not associated with a Machine.
+	BYOHAnnotation = "windowsmachineconfig.openshift.io/byoh"
+	// HostsConfigMap is the name of the ConfigMap where VMs to be configured should be described.
+	HostsConfigMap = "windows-instances"
```

Comment:
As in we create it and the user adds their data?
Wouldnt that mean that the user would be forced to use `oc edit` instead of being able to apply a config file? That seems like a large downside to me

Reply:
The user can do `oc apply -f windows-instances.yaml` and the object will get updated. This is how other OpenShift config objects work.